project-koku/koku
# COST-1488: Make internal sources cleanup script run asynchronously

**Testing**
1. Add 2 AWS sources
2. Delete a source db record to reproduce a stuck provider
3. Verify that the Provider is returned with the providers_without_sources parameter.
4. Use the uuid parmeter of the other source and verify no providers are listed.
5. Delete other source db record so both providers are stuck.
6. Cleanup only provider b62eafc0-0c6c-4879-8911-aefb227500bd
7. Cleanup other provider with no parameter given
8. Add a source and delete the source directly from the sources-api database to recreate a missing_source situation.
9. Cleanup missing source and verify it becomes a provider_without_source.

**Test Results**
[source_cleanup_async_ut.txt](https://github.com/project-koku/koku/files/6619292/source_cleanup_async_ut.txt)
